### PR TITLE
FIX: to_dict with nn models

### PR DESCRIPTION
**Pull request by martins0n**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

### CODE REVIEW

File: etna/core/mixins.py [hunk 1]
```diff
@@ -80,6 +93,22 @@ def to_dict(self):
         params["_target_"] = BaseMixin._get_target_from_class(self)
         return params
 
+    @staticmethod
+    def _unsafe_to_dict(value: Any) -> Dict[str, Any]:
+        """Collect all information about etna object in dict."""
```
Comment (Mr-Geekman) [inline etna/core/mixins.py:98]: It isn't only about etna object, we use it for callbacks from pytorch_lightning.

File: etna/core/utils.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    def wrapper(*args, **kwargs):
+        self, *args = args
+        init_args = inspect.signature(self.__init__).parameters
+
+        deepcopy_args = deepcopy(args)
+        deepcopy_kwargs = deepcopy(kwargs)
+
+        self._init_params = {}
+        args_dict = dict(
+            zip([arg for arg, param in init_args.items() if param.kind == param.POSITIONAL_OR_KEYWORD], deepcopy_args)
```
Comment (Mr-Geekman) [inline etna/core/utils.py:20]: Can we meet problems with other kinds? E.g. `kwargs` parameters: [docs](https://docs.python.org/3/library/inspect.html#inspect.Parameter.kind).

Comment (martins0n) [reply etna/core/utils.py:20]: The will be in kwargs

Comment (Mr-Geekman) [reply etna/core/utils.py:20]: Will it work with `POSITIONAL_ONLY` somehow?

Comment (martins0n) [reply etna/core/utils.py:20]: No

Comment (Mr-Geekman) [reply etna/core/utils.py:20]: I think we should write in documentation about the limitations of this approach.

Comment (martins0n) [reply etna/core/utils.py:23]: I've added comment

**Review by Mr-Geekman — APPROVED** [review]

**Pull request merged**